### PR TITLE
Hosting Command Palette: Add command to navigate to WooCommerce 

### PR DESCRIPTION
**Pull request by katinthehatsite**
Related to https://github.com/Automattic/dotcom-forge/issues/4859

## Proposed Changes

This PR adds a command to `Open WooCommerce settings`:

* If a site does not have a WooCommerce plugin activated (e.g. sites with plans below `Creator` plan or a site with a `Creator` plan but no store set up), the customer would be sent to `/woocommerce-installation/{yourAtomicSite}`;
* If a site has a WooCommerce plugin activated, the customer will be sent to WooCommerce in `wp-admin`;
* P2 sites and Jetpack sites are filtered out at the moment. @sejas should we include Jetpack sites and redirect to `wp-admin`?

This is to remain consistent with the behavior of the WooCommerce menu item at the moment.

Screenshot 2023-12-20 at 6 02 14 PM

## Testing Instructions

**To test an Atomic site with a store set up**: 

* Navigate to Calypso Live Link
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open the command palette
* Search for the `Open WooCommerce settings` command
* Select an Atomic site with the store set up 
* Confirm that you are brought to `wp-admin/admin.php?page=wc-admin`

**To test with on a site with no store**: 

* Navigate to Calypso Live Link
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open the command palette
* Search for the `Open WooCommerce settings` command
* Select a site with an Explorer plan or a Free plan
* Confirm that you are brought to `/woocommerce-installation/${ site.slug }`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 1]
```diff
@@ -1173,6 +1174,26 @@ export const useCommandsArrayWpcom = ( {
...
+			siteFunctions: {
+				onClick: ( { site, close }: { site: SiteExcerptData; close: () => void } ) => {
+					close();
+					if ( site.options?.is_wpcom_store ) {
+						navigate( `${ site.URL }/wp-admin/admin.php?page=wc-admin` );
+					} else {
+						navigate( `/woocommerce-installation/${ site.slug }` );
+					}
+				},
+				filter: ( site: SiteExcerptData ) => ! isP2Site( site ) && ! isNotAtomicJetpack( site ),
```
Comment (sejas) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx:1193]: Can we add the filterNotice:
`filterNotice: __( 'Only listing sites hosted on WordPress.com.' ),` ?

**Review by sejas — APPROVED** [review]
It looks great!
Tested with simple, atomic and it redirects to calypso. A site with woocommerce installed goes to `wp-admin`.

I added a comment about displaying the filter notice.

https://github.com/Automattic/wp-calypso/assets/779993/2054ebdc-08a1-418b-8a44-49cade950caf

**Comment by sejas** [comment]
> P2 sites and Jetpack sites are filtered out at the moment. @sejas should we include Jetpack sites and redirect to wp-admin?

Well done. It's totally fine to hide Jetpack sites, since even if we install WoCommerce there we get the `is_wpcom_store` option to false. Also the WooCommerce menu doesn't appear on the Calypso menu.

Screenshot 2023-12-21 at 11 12 28

**Comment by katinthehatsite** [comment]
Yeah makes sense, my suspicion was that we won't be able to reliably determine if a self-hosted site has WooCommerce installed or not.

File: client/sites-dashboard/components/wpcom-smp-commands.tsx (hunk 1 above)
Comment (katinthehatsite) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx:1193]: done - added in https://github.com/Automattic/wp-calypso/pull/85592/commits/535bd9cf5e42bf3740b2b61e840fdc5e93371dc1

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10444062

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @katinthehatsite for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10444062).

**Pull request merged**